checker/tests/lock/Issue524.java

```diff
@@ -0,0 +1,18 @@
+import org.checkerframework.checker.lock.qual.*;
+import org.checkerframework.dataflow.qual.*;
+
+//https://github.com/typetools/checker-framework/issues/524
+
+import java.util.concurrent.locks.*;
+
+// Note that this test is nondeterministic - it sometimes passes when it should fail.
+class Issue524 {
+    class MyClass { public Object field; }
+
+    void testLocalVariables() {
+        @GuardedBy("localLock") MyClass q = new MyClass();
```

Comment:
Why is the type `@GuardedBy("localLock")` allowed? I don't see a variable "localLock" in scope - at this point - and would forbid this type.